modules/gapi/perf/streaming/gapi_streaming_source_perf_tests.cpp

```diff
@@ -82,18 +89,216 @@ PERF_TEST_P_(VideoCapSourcePerfTest, TestPerformance)
+    resize_t res = get<3>(params);
+
+    std::vector<CfgParam> cfg_params {
+        CfgParam::create_implementation("MFX_IMPL_TYPE_HARDWARE"),
+    };
+
+    if (!type.empty()) {
+        cfg_params.push_back(CfgParam::create_decoder_id(type.c_str()));
+    }
+
+    if (!mode.empty()) {
+        cfg_params.push_back(CfgParam::create_acceleration_mode(mode.c_str()));
+    }
+
+    auto device_selector = std::make_shared<CfgParamDeviceSelector>(cfg_params);
+    auto source_ptr = cv::gapi::wip::make_onevpl_src(src, cfg_params, device_selector);
+
+    // create VPP preproc engine
+    auto required_frame_param = mock_network_info(res.first, res.second);
+    std::unique_ptr<VPLAccelerationPolicy> policy;
+    if (mode == "MFX_ACCEL_MODE_VIA_D3D11") {
+        policy.reset(new VPLDX11AccelerationPolicy(device_selector));
+    } else {
+        policy.reset(new VPLCPUAccelerationPolicy(device_selector));
+    }
+    VPPPreprocEngine preproc_engine(std::move(policy));
+    cv::gapi::wip::Data out;
+    TEST_CYCLE()
+    {
+        source_ptr->pull(out);
+        cv::MediaFrame frame = cv::util::get<cv::MediaFrame>(out);
+        cv::util::optional<pp_params> param = preproc_engine.is_applicable(frame);
+        pp_session sess = preproc_engine.initialize_preproc(param.value(),
+                                                            required_frame_param);
+        (void)preproc_engine.run_sync(sess, frame);
+    }
+
+    SANITY_CHECK_NOTHING();
+}
+
+INSTANTIATE_TEST_CASE_P(Streaming_Engine_PP, OneVPLSourcePerf_PP_Engine_Test,
+                        Values(source_description_preproc_t(files[0], codec[0], "", full_hd),
+                               source_description_preproc_t(files[0], codec[0], "", cif),
+                               source_description_preproc_t(files[0], codec[0], "MFX_ACCEL_MODE_VIA_D3D11", full_hd),
+                               source_description_preproc_t(files[0], codec[0], "MFX_ACCEL_MODE_VIA_D3D11", cif),
+                               source_description_preproc_t(files[1], codec[1], "", full_hd),
+                               source_description_preproc_t(files[1], codec[1], "", cif),
+                               source_description_preproc_t(files[1], codec[1], "MFX_ACCEL_MODE_VIA_D3D11",full_hd),
+                               source_description_preproc_t(files[1], codec[1], "MFX_ACCEL_MODE_VIA_D3D11",cif),
+                               source_description_preproc_t(files[2], codec[2], "", full_hd),
+                               source_description_preproc_t(files[2], codec[2], "", cif),
+                               source_description_preproc_t(files[2], codec[2], "MFX_ACCEL_MODE_VIA_D3D11", full_hd),
+                               source_description_preproc_t(files[2], codec[2], "MFX_ACCEL_MODE_VIA_D3D11", cif)));
+
+class OneVPLSourcePerf_PP_Engine_Bypass_Test : public TestPerfParams<source_description_preproc_t> {};
+
+PERF_TEST_P_(OneVPLSourcePerf_PP_Engine_Bypass_Test, TestPerformance)
```

Comment:
Is there any difference between `OneVPLSourcePerf_PP_Engine_Test` and `OneVPLSourcePerf_PP_Engine_Bypass_Test` ?

Reply:
Yes, according to it's name
bypass - means no preproc is actually happens. The one and only one additional cost is measure performance affect of `is_applicable` , `initialize_preproc` & `run_sync` execution in pass through mode to estimate performance degradation for normal-input

Reply:
I just didn't find any difference between these two tests, since they're so similar does it make sense to combine into one test and control `bypass`/`not-bypass` via test parameter

Reply:
agree, looks the same. should be merged in one only